packages/lib/server/repository/destinationCalendar.ts

```diff
@@ -36,6 +48,20 @@ export class DestinationCalendarRepository {
     });
   }
 
+  private static async findConflictingForUser(data: {
+    userId: number;
+    integration: string;
+    externalId: string;
+  }) {
+    return await DestinationCalendarRepository.find({
+      where: {
+        userId: data.userId,
+        integration: data.integration,
+        externalId: data.externalId,
+      },
```

Comment:
Even though right now either userId or eventTypeId is set for DestinationCalendar, it could be better to add an explicity filter here for eventTypeId:null, so that we don't accidentally consider eventType destination calendar in future.

Reply:
Good idea - implemented in [refactor: exclude eventTypeId from user calendar search](https://github.com/calcom/cal.com/pull/22286/commits/0b8242dc29c98670e73db31a5ff43369a9a63e7a)